test/e2e/specs/editor/blocks/image.spec.js

```diff
@@ -0,0 +1,549 @@
+	} );
+
+	test( 'should replace, reset size, and keep selection', async ( {
+		page,
+		pageUtils,
+		imageBlockUtils,
+	} ) => {
+		await pageUtils.insertBlock( { name: 'core/image' } );
+
+		const imageBlock = page.locator(
+			'role=document[name="Block: Image"i]'
+		);
+		const image = imageBlock.locator( 'role=img' );
+
+		const filename = await imageBlockUtils.upload(
+			imageBlock.locator( 'data-testid=form-file-upload-input' )
+		);
+
+		{
+			await expect( image ).toBeVisible();
+			await expect( image ).toHaveAttribute(
+				'src',
+				new RegExp( filename )
+			);
+
+			const regex = new RegExp(
+				`<!-- wp:image {"id":(\\d+),"sizeSlug":"full","linkDestination":"none"} -->
+<figure class="wp-block-image size-full"><img src="[^"]+\\/${ filename }\\.png" alt="" class="wp-image-\\1"/></figure>
+<!-- \\/wp:image -->`
+			);
+			expect( await pageUtils.getEditedPostContent() ).toMatch( regex );
+		}
+
+		{
+			await pageUtils.openDocumentSettingsSidebar();
+			await page.click(
+				'role=group[name="Image size presets"i] >> role=button[name="25%"i]'
+			);
+
+			await expect( image ).toHaveCSS( 'width', '3px' );
+			await expect( image ).toHaveCSS( 'height', '3px' );
+
+			const regex = new RegExp(
+				`<!-- wp:image {"id":(\\d+),"width":3,"height":3,"sizeSlug":"full","linkDestination":"none"} -->
+<figure class="wp-block-image size-full is-resized"><img src="[^"]+\\/${ filename }\\.png" alt="" class="wp-image-\\1" width="3" height="3"\\/><\\/figure>
+<!-- /wp:image -->`
+			);
+
+			expect( await pageUtils.getEditedPostContent() ).toMatch( regex );
+		}
+
+		{
+			await pageUtils.showBlockToolbar();
+			await page.click( 'role=button[name="Replace"i]' );
+
+			const replacedFilename = await imageBlockUtils.upload(
+				page
+					// Ideally the menu should have the name of "Replace" but is currently missing.
+					// Hence, we fallback to using CSS classname instead.
+					.locator( '.block-editor-media-replace-flow__options' )
+					.locator( 'data-testid=form-file-upload-input' )
+			);
+
+			await expect( image ).toHaveAttribute(
+				'src',
+				new RegExp( replacedFilename )
+			);
+			await expect( image ).toBeVisible();
+
+			const regex = new RegExp(
+				`<!-- wp:image {"id":(\\d+),"sizeSlug":"full","linkDestination":"none"} -->
+<figure class="wp-block-image size-full"><img src="[^"]+\\/${ replacedFilename }\\.png" alt="" class="wp-image-\\1"/></figure>
+<!-- \\/wp:image -->`
+			);
+			expect( await pageUtils.getEditedPostContent() ).toMatch( regex );
+		}
+
+		{
+			// Focus outside the block to avoid the image caption being selected
+			// It can happen on CI specially.
+			await page.click( 'role=textbox[name="Add title"i]' );
+			await image.click();
+			await page.keyboard.press( 'Backspace' );
+
+			expect( await pageUtils.getEditedPostContent() ).toBe( '' );
+		}
+	} );
+
+	test( 'should place caret at end of caption after merging empty paragraph', async ( {
```

Comment:
This test was originally skipped (https://github.com/WordPress/gutenberg/issues/32872), but I couldn't reproduce it after the migration. Let's enable it and see if it fixes the issue.